### PR TITLE
fix(window_state_context.js): prevent restoring if already minimized.

### PR DESCRIPTION
**Pull request by marcoscaceres**
Line 8 in particular fixed some tests that were accidentally restoring multiple times. 

The rest is just style fixes.

### CODE REVIEW

File: page-visibility/resources/window_state_context.js [hunk 1]
```diff
@@ -16,22 +17,25 @@ function window_state_context(t) {
...
 
     async function restoreAndWait() {
       const promise = visibilityEventPromise();
       await Promise.all([restore(), promise]);
-      await new Promise(resolve => t.step_timeout(resolve, 0));
+      await new Promise((resolve) => t.step_timeout(resolve, 0));
     }
 
-    return {minimize, restore, minimizeAndWait, restoreAndWait};
+  return { minimize, restore, minimizeAndWait, restoreAndWait };
```
Comment (marcoscaceres) [inline page-visibility/resources/window_state_context.js:40]:
```suggestion
    return { minimize, restore, minimizeAndWait, restoreAndWait };
```

File: page-visibility/resources/window_state_context.js [hunk 2]
```diff
@@ -2,12 +2,13 @@ function window_state_context(t) {
     let rect = null;
     let state = "restored";
     t.add_cleanup(async () => {
-        if (state === "minimized")
-            await restore();
+      if (state === "minimized") await restore();
```
Comment (noamr) [inline page-visibility/resources/window_state_context.js:5]: New line after if

File: page-visibility/resources/window_state_context.js [hunk 3]
```diff
@@ -2,12 +2,13 @@ function window_state_context(t) {
...
     });
     async function restore() {
-        state = "restored";
+        if (state !== "minimized") return;
```
Comment (noamr) [inline page-visibility/resources/window_state_context.js:8]: Ditto

**Review by noamr — APPROVED** [review]
LGTM%nit

**Comment by marcoscaceres** [comment]
@noamr, it is cool if I just prettier.js this file? The whitespace is all over the place.

**Comment by noamr** [comment]
> @noamr, it is cool if I just prettier.js this file? The whitespace is all over the place.

Yep, go nuts

**Pull request merged**